src/smallbox.rs

```diff
@@ -805,43 +762,6 @@ mod tests {
         }
     }
 
-    #[test]
-    fn test_overaligned_zst() {
```

Comment:
I think this case is not related to null pointer optimization.

Reply:
Indeed. I had reverted #47 because I had thought it was fixing an issue caused by #46. It looks like the issue #47 fixes issues which predate #46, so I'll need to redo this set of reverts.